tests/storage/test_user_directory.py

```diff
@@ -11,9 +11,319 @@
+            )
+
+    def test_populates_local_users(self):
+        u1 = self.register_user("user1", "pass")
+        u1_token = self.login(u1, "pass")
+        u2 = self.register_user("user2", "pass")
+        u2_token = self.login(u2, "pass")
+        u3 = self.register_user("user3", "pass")
+        u3_token = self.login(u3, "pass")
+
+        room = self.helper.create_room_as(u1, is_public=True, tok=u1_token)
+        self.helper.invite(room, src=u1, targ=u2, tok=u1_token)
+        self.helper.join(room, user=u2, tok=u2_token)
+
+        private_room = self.helper.create_room_as(u1, is_public=False, tok=u1_token)
+        self.helper.invite(private_room, src=u1, targ=u3, tok=u1_token)
+        self.helper.join(private_room, user=u3, tok=u3_token)
+
+        self._purge_and_rebuild_user_dir()
+
+        shares_private = self.get_users_who_share_private_rooms()
+        public_users = self.get_users_in_public_rooms()
+
+        # User 1 and User 2 are in the same public room
+        self.assertEqual(set(public_users), {(u1, room), (u2, room)})
+
+        # User 1 and User 3 share private rooms
+        self.assertEqual(
+            self._compress_shared(shares_private),
+            {(u1, u3, private_room), (u3, u1, private_room)},
+        )
+
+        # All three should have entries in the directory
+        self.assertEqual(set(self.get_users_in_user_directory().keys()), {u1, u2, u3})
+
+    def test_populates_users_in_zero_rooms(self):
+        billy_no_mates = self.register_user("user2", "pass")
+        self._purge_and_rebuild_user_dir()
+        self.assertEqual(self.get_users_in_user_directory().keys(), {billy_no_mates})
+        self.assertEqual(self.get_users_in_public_rooms(), [])
+        self.assertEqual(self.get_users_who_share_private_rooms(), [])
+
+    def test_population_excludes_support_user(self):
+        support = "@support1:test"
+        self.get_success(
+            self.store.register_user(
+                user_id=support, password_hash=None, user_type=UserTypes.SUPPORT
+            )
+        )
+
+        self._purge_and_rebuild_user_dir()
+        # TODO add support user to a public and private room. Check that
+        # users_in_public_rooms and users_who_share_private_rooms is empty.
+        self.assertEqual(self.get_users_in_user_directory(), {})
+
+    def test_population_excludes_appservice_user(self):
+        as_token = "i_am_an_app_service"
+        appservice = ApplicationService(
+            as_token,
+            self.hs.config.server_name,
+            id="1234",
+            namespaces={"users": [{"regex": r"@as_user.*", "exclusive": True}]},
+            sender="@as:test",
+        )
+        self.store.services_cache.append(appservice)
+
+        as_user = "@as_user_potato:test"
+        self.get_success(self.store.register_user(user_id=as_user, password_hash=None))
+
+        # TODO can we configure the app service up front somehow? This is a hack.
```

Comment:
would have thought it should be doable. There are `@override_config` annotations for the standard config, but not sure what the equivalent is for appservices...

Looks like you have to create files and configure those, see (& copy?) `ApplicationServiceStoreTestCase.setUp`...